vignettes/lock_and_prop.Rmd

```diff
@@ -0,0 +1,97 @@
+3. A breaking bug is discovered in a dependent package that breaks backward compatibility for that package with `{admiral}`.
+4. A dependency is no longer used in `{admiral}`.
+5. A dependency package cause an issue in R checks or R-CMD checks to fail .
+
+The first step for requesting an update for a R package in the `renv.lock` file is to raise it in our [slack channel](https://app.slack.com/client/T028PB489D3/C028SJ83KM1).  The next step is to make an issue for it in the [admiralci repository](https://github.com/pharmaverse/admiralci/) and tag `{admiral}` leads.  This will create discussion around why this update is needed. The `admiralci` leads will then make the final decision (**almost always in the affirmative**) and we can move this to the attention of other admiral extension package leads.  Once all leads are aware of this needed update, then the `renv.lock` file for the entire admiral family packages will be updated through a **centrally controlled propagation processes**.  It is expected that all repositories adopt this update `renv.lock` file.
+
+## Propagation Process
+
+Through GitHub Actions we have developed the ability to **propagate** updated lock files with our preferred versions of R and R packages throughout the family of admiral packages.  This ensures that developers working on `{admiral.test}` and `{admiralonco}` have the same common environment as those working on `{admiraldev}` and `{admiralvaccines}`. 
+
+
+The file that does the propagation process is called [`renv.yml`](https://github.com/pharmaverse/admiralci/blob/main/.github/workflows/renv.yml) located in the `.github/workflows` folder of the [admiralci repository](https://github.com/pharmaverse/admiralci/). 
+
+The process is implemented to generate `renv.lock` file automatically. It will be done for all supported R releases that need to be compatible with admiral. The propagation occurs in the following steps:
+
+1. The modification of `r-renv-lock.yml` triggers regenerate the package dependencies that automatically commit changes for `renv.lock`.
+2. A Pull Request will be created in all admiral family repos as seen below with the tile **Propagate renv.lock from pharmaverse/admiralci** title.  
+3. A `pharmaverse-bot` is the author of the Pull Request as seen in `[2]` in the image below.  
+4. The bot assign relevant reviewers to the Pull Requests. The reviewers tend to be admins or core contributors to the repo and can be adjusted if needed.  
+
+**Note:** In `[1]` you will see that one reviewer has a special shield symbol next to their name.  This person is a codeowner of the `common.yml` and is alerted whenever a change is made to this file.  You can read more about codeowners on [About Code Owners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) in GitHub Documentation
+
+
+```{r, echo = FALSE}
+knitr::include_graphics("./prop.png", dpi = 144)
+```
+
```

Comment:
Could we add something regarding what to do when the new `renv.lock` file needs to be modified because it is causing issues in some of the admiral repositories. I assume in this case a comment should be added to the issue or PR in admiralci.

Maybe it would be helpful to add this information also to the generated PRs (with a link to the issue/PR which generated the PRs).